### PR TITLE
Create dialog

### PR DESCRIPTION
**Pull request by hoangdat**
@sherlockvn It is good at Repository + Datasource + DI, please complete the presentation layer.
Thanks and BRs

### CODE REVIEW

**Comment by sherlockvn** [comment]
https://github.com/linagora/twake-on-matrix/assets/43041967/5472ab18-f05a-4822-a249-e6edcccb3a74

File: lib/data/network/interceptor/authorization_interceptor.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  String? _accessToken;
+
+  set accessToken(String? accessToken) {
+    _accessToken = accessToken;
+  }
+
+  @override
+  void onRequest(RequestOptions options, RequestInterceptorHandler handler) {
+    options.headers[HttpHeaders.authorizationHeader] = _bearerToken;
+    debugPrint('accessToken: $_bearerToken');
```
Comment (hoangdat) [inline lib/data/network/interceptor/authorization_interceptor.dart:18]: please change to use `Logs().d`

File: lib/data/datasource_impl/contact/tom_contacts_datasource_impl.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import 'package:fluffychat/data/datasource/tom_contacts_datasource.dart';
+import 'package:fluffychat/data/network/contact/tom_contact_api.dart';
+import 'package:fluffychat/domain/model/contact/contact.dart';
+import 'package:fluffychat/domain/model/contact/contact_query.dart';
+import 'package:fluffychat/domain/model/contact/contact_status.dart';
+import 'package:fluffychat/domain/model/extensions/tom_contact_extension.dart';
+
+class TomContactsDatasourceImpl implements TomContactsDatasource {
+
+  final TomContactAPI tomContactAPI;
```
Comment (hoangdat) [inline lib/data/datasource_impl/contact/tom_contacts_datasource_impl.dart:10]:
```suggestion
  final TomContactAPI tomContactAPI = getIt.get<TomContactAPI>();
```

File: lib/di/global/network_di.dart [hunk 3]
```diff
@@ -23,16 +24,15 @@ class NetworkDI extends BaseDI {
...
 
   void _bindBaseOption(GetIt get) {
-    final headers = <String, dynamic>{
-      HttpHeaders.acceptHeader: acceptHeaderDefault,
-      HttpHeaders.contentTypeHeader: contentTypeHeaderDefault
-    };
+    // final headers = {
+    //   HttpHeaders.acceptHeader: acceptHeaderDefault,
+    //   HttpHeaders.contentTypeHeader: contentTypeHeaderDefault
+    // };
```
Comment (hoangdat) [inline lib/di/global/network_di.dart:33]: IMO, you should not comment it,

File: lib/widgets/twake_components/twake_fab.dart [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+class TwakeFloatingActionButton extends StatelessWidget {
```
Comment (hoangdat) [inline lib/widgets/twake_components/twake_fab.dart:4]: Why we need to use it? how about the built-in Fab?

File: lib/domain/model/extensions/contact/list_presentation_contact_extension.dart
Comment (hoangdat) [inline lib/domain/model/extensions/contact/list_presentation_contact_extension.dart:1]: remove it. Not use anywhere

File: lib/pages/new_private_chat/widget/new_private_chat_appbar.dart [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      ),),
+    actions: isSearchBarShow
+      ? [
+        TwakeIconButton(
+          onPressed: () {
+            setState(() => isSearchBarShow = false);
+            widget.newPrivateChatController.searchKeyword = "";
+            widget.newPrivateChatController.fetchCurrentTomContacts();
+          }, 
+          tooltip: "Close",
```
Comment (imGok) [inline lib/pages/new_private_chat/widget/new_private_chat_appbar.dart:103]: We don't have intl for this ?

File: lib/widgets/twake_components/twake_floating_action_button.dart [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import 'package:flutter/material.dart';
+
+class TwakeFloatingActionButton extends StatelessWidget {
+  const TwakeFloatingActionButton({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      
+    );
```
Comment (imGok) [inline lib/widgets/twake_components/twake_floating_action_button.dart:10]: What is it supposed to do ?

**Review by sherlockvn — APPROVED** [review]

**Pull request merged**